src/form/acp.jl

```diff
@@ -45,30 +45,33 @@ end
 
 """
 ```
-sum(p[a] for a in bus_arcs) + sum(p_dc[a_dc] for a_dc in bus_arcs_dc) == sum(pg[g] for g in bus_gens) - pd - gs*v^2
-sum(q[a] for a in bus_arcs) + sum(q_dc[a_dc] for a_dc in bus_arcs_dc) == sum(qg[g] for g in bus_gens) - qd + bs*v^2
+sum(p[a] for a in bus_arcs) + sum(p_dc[a_dc] for a_dc in bus_arcs_dc) == sum(pg[g] for g in bus_gens) - sum(load[d]["pd"] for d in bus_loads) - sum(shunt[s]["gs"] for s in bus_shunts)*v^2
+sum(q[a] for a in bus_arcs) + sum(q_dc[a_dc] for a_dc in bus_arcs_dc) == sum(qg[g] for g in bus_gens) - sum(load[d]["qd"] for d in bus_loads) + sum(shunt[s]["bs"] for s in bus_shunts)*v^2
 ```
 """
-function constraint_kcl_shunt(pm::GenericPowerModel{T}, n::Int, i::Int, bus_arcs, bus_arcs_dc, bus_gens, pd, qd, gs, bs) where T <: AbstractACPForm
+function constraint_kcl_shunt(pm::GenericPowerModel{T}, n::Int, i::Int, bus_arcs, bus_arcs_dc, bus_gens, bus_loads, bus_shunts) where T <: AbstractACPForm
     vm = pm.var[:nw][n][:vm][i]
     p = pm.var[:nw][n][:p]
     q = pm.var[:nw][n][:q]
     pg = pm.var[:nw][n][:pg]
     qg = pm.var[:nw][n][:qg]
     p_dc = pm.var[:nw][n][:p_dc]
     q_dc = pm.var[:nw][n][:q_dc]
+    load = pm.ref[:nw][n][:load]
+    shunt = pm.ref[:nw][n][:shunt]
```

Comment:
This is a slight abuse of the design goal of constraint templates.  The primary goal of constraint templates was to abstract away all of the data model form the constraints posting functions.  A guiding principal is that all `ref` lookups happen in the template and all `var` calls happen in the constraint implementation.

For example, instead of passing the `bus_loads` index list to this function you could instead pass something like,

```
bus_loads_values = [load[d]["pd"] for d in ref(pm, n, :bus_loads, i)]
```
At this moment it's not clear to me what the ideal design is.  Let me think over the weekend and lets meet early next week to brain storm about different options.